sound/soc/sof/amd/acp.c

```diff
@@ -481,11 +481,18 @@ int amd_sof_acp_probe(struct snd_sof_dev *sdev)
 		return -ENOMEM;
 
 	adata->dev = sdev;
+	adata->dmic_dev = platform_device_register_data(sdev->dev, "dmic-codec",
+							PLATFORM_DEVID_NONE, NULL, 0);
+	if (IS_ERR(adata->dmic_dev)) {
+		dev_err(sdev->dev, "error: failed to create DMIC device\n");
+		return PTR_ERR(adata->dmic_dev);
+	}
```

Comment:
@plbossart just out of curiosity this looks lexactly like the dmic platform device we create for HDA dev. Why do you only call this one out?

Reply:
@ranj063 AMD are suggesting to move the DMIC platform device under the SOF platform device.

What I am questioning is why there is an SOF platform device in the first place - not the use of a platform device for DMIC, as you mentioned Intel uses the same trick.

Reply:
Ahh I see what you mean. I think its only a typo. THey likely meant platform data instead

Reply:
The commit message is focused on devices and device hierarchy....

````
ASoC: SOF: amd: refactor dmic-codec platform device creation
Instead of creating device node for dmic-codec as ACP PCI device
as parent, create device node under SOF platform device.
````
It's not a typo IMHO.

Reply:
@vijendarmukunda I don't know what an "ACP SOF device" is, so can't follow the logic/explanations.

Reply:
@plbossart : Its sof_pci_dev which is created by invoking sof_pci_probe() ->snd_sof_device_probe sequence when ACP PCI driver probe is invoked.
For ex: [acp_pci_rmb_probe](https://elixir.bootlin.com/linux/latest/C/ident/acp_pci_rmb_probe) for RMB platform

Reply:
@vijendarmukunda I only see a PCI device, not a platform device, so I don't get what the change is.

Where is the 'SOF platform device' created, if it exists?

Reply:
@plbossart :  Sorry its typo error. its not SOF platform device. 

For example for RMB platform , ACP PCI driver probe will be invoked .
https://elixir.bootlin.com/linux/latest/source/sound/soc/sof/amd/pci-rmb.c#L84
From this call, it invokes sof_pci_probe() call.
https://elixir.bootlin.com/linux/latest/source/sound/soc/sof/sof-pci-dev.c#L184
From sof_pci_probe() -> it will invoke snd_sof_device_probe() which allocates snd_sof_dev structure.

https://elixir.bootlin.com/linux/latest/source/sound/soc/sof/core.c#L346

Under this snd_sof_dev device , we want to create platform devices for generic dmic codec driver. 
we will further add support for soundwire stack where we will create platform devices for soundwire manager under 
sdev device scope.

we will update the commit message.

Reply:
@ranj063 @plbossart  please help to review.
Updated log for DMIC platform device creation and updated commit message for 
"ASoC: SOF: amd: refactor dmic codec platform device creation"